### PR TITLE
style: More traversal cleanup.

### PR DESCRIPTION
**Pull request by emilio**
This is cleanup that allows me to fix https://bugzilla.mozilla.org/show_bug.cgi?id=1415013 in a more straight-forward way.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/19164)

### CODE REVIEW

**Comment by emilio** [comment]
https://treeherder.mozilla.org/#/jobs?repo=try&revision=68742a9e7b59bdd5f3750919dea7f39b27bbd86f

**Comment by emilio** [comment]
@bors-servo try

(Also, @nox or @SimonSapin may want to stamp it if they feel confident about it)

**Comment by emilio** [comment]
@bors-servo try

This should be ready for review, I had to fix a Servo bug (https://github.com/servo/servo/pull/19164/commits/39f6bb728abe21f280fbcd726c4dd8e5c9370262).

**Comment by emilio** [comment]
Those two tests would get fixed by https://github.com/servo/servo/pull/19175.

File: components/script/dom/node.rs [hunk 1]
```diff
@@ -483,6 +483,24 @@ impl Node {
...
 
+    // FIXME(emilio): This and the function below should move to Element.
+    pub fn note_dirty_descendants(&self) {
+        debug_assert!(self.is_in_doc());
+
+        // FIXME(emilio): Is there a safe way to do this without the overhead of
+        // rooting? I'm pretty sure this cannot GC!
+        let mut current = Some(DomRoot::from_ref(self));
+
+        while let Some(node) = current.take() {
```
Comment (jdm) [inline components/script/dom/node.rs]: There's no good answer for the rooting question, unfortunately. As for this code, let's use:
```rust
for node in self.ancestors() {
    if node.get_flag(...) {
        return;
    }
    node.set_flag(...);
}
```

**Review by jdm — DISMISSED** [review]

File: components/script/dom/node.rs (hunk 1 above)
Comment (emilio) [reply components/script/dom/node.rs]: I changed it to inclusive_ancestors, which is what it is intended to do.

**Comment by emilio** [comment]
Heycam's back!

r? @heycam (if @bzbarsky is busy)

**Pull request merged**